server/lib/resolvers/wrapResolverWithHooks.js

```diff
@@ -0,0 +1,36 @@
+const _ = require('lodash')
+const {log} = require('../util/logger')
+const request = require('request')
+
+exports.wrapResolverWithHooks = function wrapResolverWithHooks (resolver, resolverMapping) {
+  return wrapResolverWithMappingHooks(resolver, resolverMapping)
+}
+
+function wrapResolverWithMappingHooks (resolverFn, resolverMapping) {
+  return (obj, args, context, info) => {
+    return new Promise(async (resolve, reject) => {
+      if (resolverMapping.preHook && !_.isEmpty(resolverMapping.preHook)) {
+        log.info(resolverMapping.preHook)
+        try {
+          request.get(resolverMapping.preHook)
```

Comment:
Checked that and try catch is not going to work here. We should use 
```.on('error', function(err) {
    log.err(err)
  })
```

Reply:
We actually have very few callbacks in the code base and await/async with try/catch is the preferred way, but we don't want to block execution here. So, +1 on @wtrocki 's suggestion.